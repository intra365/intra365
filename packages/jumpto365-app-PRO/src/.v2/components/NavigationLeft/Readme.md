NavigationLeft example:

## English
```js
<NavigationLeft language="en"/>
```

## Danish
```js
<NavigationLeft language="da"/>
```

## Japanese
```js
<NavigationLeft language="ja"/>
```