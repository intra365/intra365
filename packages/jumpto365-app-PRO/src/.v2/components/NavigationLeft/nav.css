.ms-CommandBarSearch-input  {

    cursor: text !important;
}

.bluecommands {
    background-color: #0B559F;
    color: #ffffff;
}


.NavContext {
    border-bottom: 1px solid #cccccc;
    height: 41px;
    width: 100%;
}
.NavLeft {
    display:flex;
    flex-direction: column;
    background-color: #e7e7e7; 

    padding:8px;
   

}

.NavLeftGroup {
    background-color: #e7e7e7
}


.NavLeftItem {
    border-left: 3px solid #cfcfcf;    

    padding-left:8px;
    margin-left:-8px;
}

.NavLeftItem:hover {
    
    border-left: 3px solid blue;    

    padding-left:8px;
    margin-left:-8px;
}


.NavLeftItemActive {
    color: #333333 !important;
    background-color: #e7e7e7 
    
}

.NavLeftExpanded{
min-width: 280px;
height:100%;
background-color: #e7e7e7; 
}

.NavLeftCollapsed{
    max-width: 93px;
    background-color: #e7e7e7; 
    height:100%;
}

.NavLeftSettings {
    padding-top:10px;
    border-top: 1px solid #cccccc;
    background-color: #e7e7e7;
    bottom:0px;
    
    position: absolute;
}


.NavTogglerExpanded{
    margin-top:-25px;
    margin-left:250px;
    cursor: pointer;
    float: left;
}

.NavTogglerCollapsed{
margin-top:10px;
    margin-left: 13px;
    margin-bottom: 10px
}