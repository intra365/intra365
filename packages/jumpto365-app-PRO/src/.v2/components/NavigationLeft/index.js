import React, { Component } from 'react';

import "./nav.css"  

import PropTypes from 'prop-types'

import 'office-ui-fabric-react/dist/css/fabric.css'

import Jumpto365App,{getSetting} from "../_Contexts/Jumpto365App"
import { DefaultButton,IconButton } from 'office-ui-fabric-react/lib/Button';
import { navigate ,Location} from "@reach/router"
import { Subscribe } from 'react-contextual'
import { TooltipHost } from 'office-ui-fabric-react';


function getTenantSettings(context){
  
    
  
  var settings = {text:  Jumpto365App.prompt("64",context),homePath : context && context.isAuthenticated && context.me ? `/@/${context.me.domain}/-`:  "/",navigation : []}
  
  if (context.userId){
    //if (getSetting("module-personaldashboard",{enabled:false}).enabled) settings.homePath = `/@/${context.userId}`
}
  if (context && context.tenant && context.tenant.settings ){
      
      var data = context.tenant.settings 

      if ( data.name ) settings.text = data.name
      if (data.navigation ) settings.navigation = data.navigation

  }
  
  return settings

} 
class LinkTo extends Component { 
  
      render(){
        var backgroundColor = this.props.backgroundColor ? this.props.backgroundColor : "#e7e7e7"
        if (this.props.hide){
          return false
        }
        return (
<div className="NavLeftItem ms-font-m" >

{this.props.collapsed ?
<IconButton
            data-automation-id="upload"
            ariaDescription={this.props.text}
            title={this.props.text}
            iconProps={{ iconName: this.props.icon}} 
            disabled={this.props.disabled}
            style={{ backgroundColor:backgroundColor}}
            onClick={()=> {
              if (this.props.external){
                window.open(this.props.path,"_blank")
                return
              }
              
              navigate(this.props.path)}}
            />
:
<DefaultButton
            data-automation-id="upload"
            text={this.props.collapsed ? null : this.props.text}
            iconProps={{ iconName: this.props.icon}} hero
            disabled={this.props.disabled}
  icon={this.props.collapsed}
            style={{ backgroundColor:backgroundColor}}
            onClick={()=> {
              if (this.props.external){
                window.open(this.props.path,"_blank")
                return
              }
              
              
              navigate(this.props.path)}}
            />

}
            {this.props.children}
</div>
      )}
}


/**
 * Describe overall purpose of the component
 *
 * @export
 * @class NavigationLeft
 * @extends {Component}
 */
export default class NavigationLeft extends Component { 

    static propTypes  = {
        height : PropTypes.string,
        language: PropTypes.string  ,
        isCollapsed : PropTypes.any,
        tenantHome : PropTypes.string,
        tenantName: PropTypes.string,
        onCollaps : PropTypes.func,
        globalContext : PropTypes.object
    } 
    

    
        constructor(props) {
            super(props);
            
            this.state = this.getnewstate(Jumpto365App.getNavigationState())
 
          }
        

        getnewstate = (collapsed) => {
          return {navClass: collapsed ? "NavLeftCollapsed " : "NavLeftExpanded", isCollapsed:collapsed }
        }
        
        update = (collapsed) => {
         if (this.props.onCollaps){
           this.props.onCollaps(collapsed)
         }
         this.setState(this.getnewstate(collapsed))
          
        }

          setCollapsed = (collapsed) => {
            Jumpto365App.setNavigationState(collapsed)    
            this.update(collapsed)
          }
  
         
          render() {
            var context = this.props.globalContext
           

           
            function prompt(id){
               return Jumpto365App.prompt(id,context)
            } 
            var isAuthenticated = context ? context.isAuthenticated : false
        
            return (
              <Subscribe>{globalContext => (
                <Location>
                {({ location}) => (
                <div className={this.state.navClass} >
              
            
              <div  className="NavLeft" style={{height:this.props.height }}>
              <div className="NavContext">
              <NavigationLeftContext globalContext={globalContext} tenantName={this.props.tenantName} isCollapsed={this.state.isCollapsed}></NavigationLeftContext>
         </div>
              <div className="NavLeftGroup">
                   <NavigationLeftGroup globalContext={globalContext} isCollapsed={this.state.isCollapsed} navigation={this.props.navigation}></NavigationLeftGroup>
                   </div>
                   <div className="NavLeftSettings">
              <NavigationLeftSettings globalContext={globalContext} onCollapsed={()=>{this.setCollapsed(!this.state.isCollapsed)}} isCollapsed={this.state.isCollapsed}></NavigationLeftSettings>

            </div>
              </div>
              
              
              </div>)}
            </Location>
          )}
        </Subscribe>
            );
          }

 
}


export  class NavigationLeftContext extends Component { 
render() {
  var context = this.props.globalContext
            
  function prompt(id){
     return Jumpto365App.prompt(id,context)
  } 
  var isAuthenticated = context ? context.isAuthenticated : false

  
  var settings = getTenantSettings(context)
  return (
    <div >
    <LinkTo collapsed={this.props.isCollapsed} backgroundColor={this.props.backgroundColor} icon="Home" path={settings.homePath} text={settings.text} />
  
              

              </div>
  )

}
}


export  class NavigationLeftGroup extends Component { 


    



  render() {
    var context = this.props.globalContext
    var hasFilterMenuAction = context.hostSettings ?  context.hostSettings.hasFilterMenuAction : true
    var hasWTWMenuAction = context.hostSettings ?  context.hostSettings.hasWTWMenuAction : true
    var hasEmbedMenuAction = context.hostSettings ?  context.hostSettings.hasEmbedMenuAction : true
    var blogURL = context.hostSettings ?  context.hostSettings.blogURL : "https://medium.com/jumpto365"
    var helpURL = context.hostSettings ?  context.hostSettings.helpURL : "https://jumpto365.zendesk.com/hc/en-us"

    function prompt(id){
       return Jumpto365App.prompt(id,context)
    } 
    var isAuthenticated = context ? context.isAuthenticated : false

    var settings = getTenantSettings(context)

    if (this.props.navigation){
      settings.navigation = this.props.navigation
    }
    var isCollapsed = this.props.isCollapsed
    var backgroundColor = this.props.backgroundColor
    var tenantNavigation = settings.navigation.map(function(node,ix){
      return <LinkTo key={ix} collapsed={isCollapsed} icon={node.icon}  backgroundColor={backgroundColor} path={node.path}  text={node.text} />
    })
  
    return (
      <div >
              
              
              
              <LinkTo collapsed={this.props.isCollapsed}  icon="LocaleLanguage" path="/language" text="Translated versions"/>
              <LinkTo hide={!hasWTWMenuAction} collapsed={this.props.isCollapsed} icon="DeveloperTools"  backgroundColor={this.props.backgroundColor} path="/scenario/microsoft/microsoft365" text={prompt("33") /* Which tool when? */}/> {/* Repair is also a potential */}
             <LinkTo hide={!hasFilterMenuAction} collapsed={this.props.isCollapsed} icon="Filter" backgroundColor={this.props.backgroundColor} path="/pages/docs/features/licenseinformation"  text={prompt("27") /* by license */}/>
             <LinkTo hide={!hasEmbedMenuAction} collapsed={this.props.isCollapsed} icon="Embed" backgroundColor={this.props.backgroundColor} path="/embed"  text={prompt("14") /* by license */}/>
              <LinkTo  collapsed={this.props.isCollapsed} external icon="News" backgroundColor={this.props.backgroundColor} path={blogURL}  text="Blog"/>
              <LinkTo  collapsed={this.props.isCollapsed} external icon="Help" backgroundColor={this.props.backgroundColor} path={helpURL}  text="Help"/>

              {/* <LinkTo  collapsed={this.props.isCollapsed} external icon="Lightbulb" backgroundColor={this.props.backgroundColor} path="https://ideas2.jumpto365.com"  text="Ideas"/> */}
     
              
              {tenantNavigation}
              
        
   

            
              {(isAuthenticated && false) &&
              <LinkTo collapsed={this.props.isCollapsed} icon="ShareiOS" backgroundColor={this.props.backgroundColor} path="/upload"  text={prompt("42") /* upload */}/>
              }
              {/* <LinkTo collapsed={this.state.isCollapsed} icon="Embed" path="/embed" text={prompt("14")  }/>
            <LinkTo collapsed={this.state.isCollapsed} icon="LocaleLanguage" path="/language" text={prompt("6") }/> */}



            </div>
    )

  }
  }

  
export  class NavigationLeftSettings extends Component { 
  render() {
    var context = this.props.globalContext
    var aboutURL = context.hostSettings ?  context.hostSettings.aboutURL : "/about"

    function prompt(id){
       return Jumpto365App.prompt(id,context)
    } 
    var isAuthenticated = context ? context.isAuthenticated : false
return (

  <div >
  {/* <LinkTo collapsed={this.props.isCollapsed} icon="JoinOnlineMeeting" backgroundColor={this.props.backgroundColor} path="/pages/docs/insiders" text="Insiders" /> */}
  <LinkTo collapsed={this.props.isCollapsed} icon="Info" backgroundColor={this.props.backgroundColor} path={aboutURL} text={prompt("19")} />
  {/* <LinkTo collapsed={this.props.isCollapsed} icon="Contact" backgroundColor={this.props.backgroundColor} path="/login" text={prompt("38")}/> */}
  {/* <LinkTo collapsed={this.state.isCollapsed} icon="Settings" path="/user" text="Settings"/> */}
  {this.props.onCollapsed &&
  <i  onClick={()=>{
    this.props.onCollapsed()}} class={`NavToggler${this.props.isCollapsed ? "Collapsed": "Expanded"} ms-Icon ms-Icon--DoubleChevron${this.props.isCollapsed ? "Right": "Left"} ms-font-size-m `} style={{color:"black"}} aria-hidden="true"></i>
}
  </div>
)
  }
  }
  