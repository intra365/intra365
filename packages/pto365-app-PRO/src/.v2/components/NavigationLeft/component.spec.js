import React, { Component } from 'react';

import NavigationLeft from '.'

test('Check that it renders', () => {

    const wrapper = shallow(<NavigationLeft/>);

    expect(wrapper).toMatchSnapshot();
 
}

)