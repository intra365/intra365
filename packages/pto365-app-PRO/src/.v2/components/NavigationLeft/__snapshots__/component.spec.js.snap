// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check that it renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavigationLeft />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="NavLeft"
        style={
                Object {
                        "height": undefined,
                      }
        }
>
        <div
                className="NavContext"
        >
                <NavigationLeftContext
                        globalContext={undefined}
                        isCollapsed={false}
                        tenantName={undefined}
                />
        </div>
        <div
                className="NavLeftGroup"
        >
                <NavigationLeftGroup
                        globalContext={undefined}
                        isCollapsed={false}
                />
        </div>
        <div
                className="NavLeftSettings"
        >
                <NavigationLeftSettings
                        globalContext={undefined}
                        isCollapsed={false}
                        onCollapsed={[Function]}
                />
        </div>
</div>,
      "className": "NavLeftExpanded",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="NavContext"
>
            <NavigationLeftContext
                        globalContext={undefined}
                        isCollapsed={false}
                        tenantName={undefined}
            />
</div>,
          <div
            className="NavLeftGroup"
>
            <NavigationLeftGroup
                        globalContext={undefined}
                        isCollapsed={false}
            />
</div>,
          <div
            className="NavLeftSettings"
>
            <NavigationLeftSettings
                        globalContext={undefined}
                        isCollapsed={false}
                        onCollapsed={[Function]}
            />
</div>,
        ],
        "className": "NavLeft",
        "style": Object {
          "height": undefined,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavigationLeftContext
              globalContext={undefined}
              isCollapsed={false}
              tenantName={undefined}
/>,
            "className": "NavContext",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "globalContext": undefined,
              "isCollapsed": false,
              "tenantName": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavigationLeftGroup
              globalContext={undefined}
              isCollapsed={false}
/>,
            "className": "NavLeftGroup",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "globalContext": undefined,
              "isCollapsed": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavigationLeftSettings
              globalContext={undefined}
              isCollapsed={false}
              onCollapsed={[Function]}
/>,
            "className": "NavLeftSettings",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "globalContext": undefined,
              "isCollapsed": false,
              "onCollapsed": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="NavLeft"
          style={
                    Object {
                              "height": undefined,
                            }
          }
>
          <div
                    className="NavContext"
          >
                    <NavigationLeftContext
                              globalContext={undefined}
                              isCollapsed={false}
                              tenantName={undefined}
                    />
          </div>
          <div
                    className="NavLeftGroup"
          >
                    <NavigationLeftGroup
                              globalContext={undefined}
                              isCollapsed={false}
                    />
          </div>
          <div
                    className="NavLeftSettings"
          >
                    <NavigationLeftSettings
                              globalContext={undefined}
                              isCollapsed={false}
                              onCollapsed={[Function]}
                    />
          </div>
</div>,
        "className": "NavLeftExpanded",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="NavContext"
>
              <NavigationLeftContext
                            globalContext={undefined}
                            isCollapsed={false}
                            tenantName={undefined}
              />
</div>,
            <div
              className="NavLeftGroup"
>
              <NavigationLeftGroup
                            globalContext={undefined}
                            isCollapsed={false}
              />
</div>,
            <div
              className="NavLeftSettings"
>
              <NavigationLeftSettings
                            globalContext={undefined}
                            isCollapsed={false}
                            onCollapsed={[Function]}
              />
</div>,
          ],
          "className": "NavLeft",
          "style": Object {
            "height": undefined,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavigationLeftContext
                globalContext={undefined}
                isCollapsed={false}
                tenantName={undefined}
/>,
              "className": "NavContext",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "globalContext": undefined,
                "isCollapsed": false,
                "tenantName": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavigationLeftGroup
                globalContext={undefined}
                isCollapsed={false}
/>,
              "className": "NavLeftGroup",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "globalContext": undefined,
                "isCollapsed": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavigationLeftSettings
                globalContext={undefined}
                isCollapsed={false}
                onCollapsed={[Function]}
/>,
              "className": "NavLeftSettings",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "globalContext": undefined,
                "isCollapsed": false,
                "onCollapsed": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
